src/Servers.jl

```diff
@@ -119,15 +135,47 @@ e.g.
         startwrite(http)
         write(http, "response body")
         write(http, "more response body")
-        return
     end
+```
+
+The `server=` option can be used to pass an already listening socket to
+`HTTP.listen`. This allows control of server shutdown.
 
-    # pass in own server socket to control shutdown
+e.g.
+```
     server = Sockets.listen(Sockets.InetAddr(parse(IPAddr, host), port))
     @async HTTP.listen(f, host, port; server=server)
-    # close server which will stop HTTP.listen
+
+    # Closeing server will stop HTTP.listen.
     close(server)
 ```
+
+To run the following HTTP chat example, open two Julia REPL windows and paste
+the example code into both of them. Then in one window run `chat_server()` and
+in the other run `chat_client()`, then type `hello` and press return.
+Whatever you type on the client will be displayed on the server and vis-versa.
+
+```
+using HTTP
+
+function chat(io::HTTP.Stream)
```

Comment:
This is a great example

Reply:
Chat? Real-time dashboards? 😅

Reply:
No, I know that WebSockets is really useful for building interactive browser integrated stuff.
It's a rhetorical question really.
WebSockets is necessary because that is what browsers have a javascript API for. But it's really a bit of a historical oddity due to the fact that the bi-directional streaming that is already built into the HTTP/1.1 spec was not initially widely (or well) implemented.
As this chat example demonstrates, it is perfectly possible to use a plain old HTTP/1.1 connection as a bi-directional socket. And in some cases, if browser interaction is not required, it will be more efficient.